table.go

```diff
@@ -47,31 +57,133 @@ type RoutingTable struct {
 	maxLatency time.Duration
 
 	// kBuckets define all the fingers to other nodes.
-	Buckets    []*Bucket
+	Buckets    []*bucket
 	bucketsize int
 
 	cplRefreshLk   sync.RWMutex
 	cplRefreshedAt map[uint]time.Time
 
+	// replacement candidates for a Cpl
+	cplReplacementCache *cplReplacementCache
+
 	// notification functions
 	PeerRemoved func(peer.ID)
 	PeerAdded   func(peer.ID)
+
+	// function to determine the validity of a peer for RT membership
+	PeerValidationFnc PeerValidationFnc
```

Comment:
Should this really be public?

Reply:
@Stebalien 

I'd make it public for some easy testing but have fixed this now.